pypck/pck_commands.py

```diff
@@ -74,6 +74,14 @@ class PckParser:
         r"(?P<block_id>\d)(?P<text>.{0,12})"
     )
 
+    # Pattern to parse the static and dynamic group membership status
+    PATTERN_STATUS_GROUPS = re.compile(
+        r"=M(?P<seg_id>\d{3})(?P<mod_id>\d{3})\.G(?P<kind>[DP])"
+        r"(?:(?P<g1>\d{3}))?(?:(?P<g2>\d{3}))?(?:(?P<g3>\d{3}))?(?:(?P<g4>\d{3}))?"
+        r"(?:(?P<g5>\d{3}))?(?:(?P<g6>\d{3}))?(?:(?P<g7>\d{3}))?(?:(?P<g8>\d{3}))?"
+        r"(?:(?P<g9>\d{3}))?(?:(?P<g10>\d{3}))?(?:(?P<g11>\d{3}))?(?:(?P<g12>\d{3}))?"
+    )
```

Comment:
Unfortunately, I realized that this pattern is **not** correct.
After GP/GD there must be the number of groups as a three-digit value. As far as I can see this is always 12 for GP and 8 for GD (no matter how many groups are actually programmed).
However, you should add `(?P<groups>\d{3})` at the end of first line in the expression and update everything else accordingly.
Sorry for that!

Reply:
Ah ... I was already thinking something is special about groups 8, 10, 12 or something is weird about my modules. Most of my UP modules have 8 most of my din rail module have 10 as the number of groups.  The number of static groups is indeed 12 for all of them.